### PR TITLE
added a warning prompt when a user wishes to download a potentially dangerous file

### LINKED ISSUE #2135: Produce a warning prompt when users wishes to download a potentially dangerous file
Description:
Currently, when a user downloads a potentially dangerous file type, the app allows them to do so without any warning.

For improvement, we can add a user warning just when the user decides to download a potentially dangerous file. For the android system, some are: .apk, .appx, .appxbundle -- but could be changed. I could gather more information regarding this one.

Reference: https://github.com/FlowCrypt/flowcrypt-security/issues/224

### PR DESCRIPTION
**Pull request by DenBond7**
This PR added a warning prompt when a user wishes to download a potentially dangerous file

close #2135

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities

### CODE REVIEW

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/AttachmentInfo.kt [hunk 1]
```diff
@@ -183,5 +183,11 @@ data class AttachmentInfo(
   companion object {
     const val DEPTH_SEPARATOR = "/"
     const val INNER_ATTACHMENT_PREFIX = "inner_"
+
+    val DANGEROUS_FILE_EXTENSIONS = arrayOf(
+      "apk",
+      "appx",
+      "appxbundle"
+    )
```
Comment (sosnovsky) [inline FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/AttachmentInfo.kt:191]: will it be safe to open .js file on Android? (will it be opened in browser or just in text editor?)

**Review by sosnovsky — APPROVED** [review]
Works well 👍

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/AttachmentInfo.kt (hunk 1 above)
Comment (DenBond7) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/AttachmentInfo.kt:191]: Good question. I've tested it on an emulator and the system can't open such a file.

image

Comment (sosnovsky) [reply FlowCrypt/src/main/java/com/flowcrypt/email/api/email/model/AttachmentInfo.kt:191]: Looks good 👍

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**